delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/model/resolvers/ResolverValue.scala

```diff
@@ -68,6 +71,36 @@ object ResolverValue {
     override def tpe: ResolverType = ResolverType.CrossProject
   }
 
+  /**
+    * Generate the value payload from an id and a value
+    * @param id             the id of the resolver
+    * @param resolverValue  the value to encode as Json
+    */
+  def generatePayload(id: Iri, resolverValue: ResolverValue): Json = {
```

Comment:
why do you need something like this? Don't you have already a way to convert your `ResolverValue` to JsonLd using the `JsonLdDecoder[ResolverValue]`. Seems to me that the same has to be used

Reply:
I see you don't have it for `ResolverValue` but for `Resolver`. Then convert your `ResolverValue` to `Resolver` and use the compacted for

Reply:
It is the Resolver (and not the ResolverValue) which is exposed in the api. But you are right, some refactorings can be made to make things better.

Reply:
Since you are talking about the Json value, does not matter if it is `Resolver` or `ResolverValue`, since the way it is serialized to the client should be the same, as far as I'm aware.